core/sysparam.c

```diff
@@ -719,97 +761,65 @@ sysparam_status_t sysparam_get_string(const char *key, char **destptr) {
     return SYSPARAM_OK;
 }
 
-sysparam_status_t sysparam_get_int(const char *key, int32_t *result) {
-    char *buffer;
-    char *endptr;
+sysparam_status_t sysparam_get_int32(const char *key, int32_t *result) {
     int32_t value;
+    size_t actual_length;
+    bool is_binary;
     sysparam_status_t status;
 
-    status = sysparam_get_string(key, &buffer);
+    status = sysparam_get_data_static(key, (uint8_t *)&value, sizeof(int32_t),
+                                      &actual_length, &is_binary);
     if (status != SYSPARAM_OK) return status;
-    value = strtol(buffer, &endptr, 0);
-    if (*endptr) {
-        // There was extra crap at the end of the string.
-        free(buffer);
+    if (!is_binary || actual_length != sizeof(int32_t))
         return SYSPARAM_PARSEFAILED;
-    }
-
     *result = value;
-    free(buffer);
-    return SYSPARAM_OK;
+    return status;
 }
 
-sysparam_status_t sysparam_get_bool(const char *key, bool *result) {
-    char *buffer;
+sysparam_status_t sysparam_get_int8(const char *key, int8_t *result) {
```

Comment:
Do we really need a get_bool that parses a string parameter?

With a string we need to malloc the result, but with the proposed get_int8 there is no need to malloc so it might be safer in more places. Patched the reading of the payload so it can be unaligned for this too. With get_uint8 a bool version seems redundant as a byte is the smallest unit of storage.

This is all just wip as I integrate use of sysparam in some code, and it has been really helpful, a big thank you to @foogod